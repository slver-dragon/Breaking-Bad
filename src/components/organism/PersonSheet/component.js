import React from "react";
import style from "./style.module.scss";
import Error from "../../atom/Error";
import Loader from "../../atom/Loader";

let statusStyle = "";
const PersonSheet = ({ character, randomQuote, isLoading, errorValue }) => {
  switch (character.status) {
    case "Alive": {
      statusStyle = style.alive;
      break;
    }
    case "Deceased": {
      statusStyle = style.deceased;
      break;
    }
    default: {
      statusStyle = style.presumedDead;
    }
  }
  const statusElement = (
    <p className={style.state + " " + statusStyle}>{character.status}</p>
  );

  return (
    <div className={style.wrapper}>
      {errorValue ? (
        <Error textError={"Ошибка загрузки данных: " + errorValue} />
      ) : !isLoading ? (
        <div className={style.wrapper}>
          <div className={style.main}>
        <div className={style.photoWrapper}>
          <img
            className={style.photo}
            src={character.img}
            alt={character.name}
          />
        </div>
        <div className={style.inform}>
          {statusElement}
          <p className={style.name}>{character.name}</p>
          <p>
            Дата рождения:{" "}
            <span className={style.birthday}>{character.birthday} </span>
          </p>
          <p>
            Ник-нейм:{" "}
            <span className={style.nickname}>{character.nickname} </span>
          </p>
          <p className={style.quoteMargin}>
            Цитата: <span className={style.quote}>{randomQuote} </span>
          </p>
        </div>
      </div>  
        </div>
      ) : (
        <Loader />
      )}
      
    </div>
  );
};

export default PersonSheet;