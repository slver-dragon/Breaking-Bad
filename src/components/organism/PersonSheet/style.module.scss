@import "../../../assets/styles/style";

.main {
  display: flex;
  margin: 3rem 0;
  .photoWrapper {
    width: 36.5rem;
    height: 36.5rem;
    object-fit: cover;
    margin-right: 7vw;
    overflow-y: hidden;
    .photo {
      width: 100%;
      height: auto;
    }
  }
  .inform {
    color: $lightFontcolor;
    max-width: 35.96%;
    .state {
      display: inline-block;
      padding: 0.5rem 0.75rem;
      margin: 0rem 0 2rem;
    }
    .alive {
      @include stateOnColor;
    }
    .deceased {
      @include stateOffColor;
    }
    .presumedDead {
      @include statePDColor;
    }
    .name {
      @include personPageBigFont;
      margin-bottom: 1rem;
      color: $black;
    }
    .birthday,
    .nickname,
    .quote {
      color: $black;
    }
    .quoteMargin {
      margin-top: 2rem;
    }
    p {
      margin-bottom: 1rem;
    }
  }
}
