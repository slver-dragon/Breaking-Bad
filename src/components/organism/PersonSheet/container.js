import { useSelector } from "react-redux";
import { useParams } from "react-router-dom";
import Error from "../../atom/Error";
import PersonSheet from "./component";

export const PersonSheetContainer = () => {
  let character = "";
  const { id } = useParams();
  const characters = useSelector((state) => state.personData.characters);
  for (let i = 0; i < characters.length; i++) {
    if (String(characters[i].char_id) === id) {
      character = characters[i];
      break;
    }
  }

  return (
    <div>
      {character ? (
        <PersonSheet character={character} />
      ) : (
        <Error textError={"Отсутствие данных персонажа."} />
      )}
    </div>
  );
};

export const container = PersonSheetContainer;
