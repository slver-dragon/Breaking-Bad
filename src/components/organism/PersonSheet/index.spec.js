import React from "react";
import { shallow } from "enzyme";
// import { PersonSheet, PersonListContainer } from ".";
import PersonSheet from "./component";
// import PersonListContainer, { container } from "./container";

describe("Testing organism PersonSheet:", () => {
  let component = null;
  //   const onBtnClick = jest.fn();
  //   const onClickEnter = jest.fn();
  const setComponent = (props) => shallow(<PersonSheet {...props} />);
  // const setContainer = () => shallow(<PersonListContainer />);

  // test("container is rendering", () => {
  //   container = setContainer();
  //   console.log(container.debug())
  //   expect(container).toMatchSnapshot();
  // });

  test("component is rendering", () => {
    component = setComponent({character: {status: "Alive" }});
    expect(component).toMatchSnapshot();
  });

  test("status change", () => {
    component = setComponent({character: {status: "Alive"}});
    expect(component.find(".alive").length).toBe(1);
    component = setComponent({character: {status: "Deceased"}});
    expect(component.find(".deceased").length).toBe(1);
    component = setComponent({character: {status: "Presumed Dead"}});
    expect(component.find(".presumedDead").length).toBe(1);
  });

  test("error output", () => {
    component = setComponent({
      character: { status: "Alive" },
      errorValue: "Test error",
    });
    expect(component.find("ErrorMessage").length).toBe(1);
  });

  test("Loader output", () => {
    component = setComponent({
      character: { status: "Alive" },
      isLoading: true,
    });
    expect(component.find("Loader").length).toBe(1);
  });
});
