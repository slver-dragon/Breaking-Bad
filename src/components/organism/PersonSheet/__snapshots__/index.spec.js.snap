// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing organism PersonSheet: component is rendering 1`] = `
<div
  className="wrapper"
>
  <div
    className="wrapper"
  >
    <div
      className="main"
    >
      <div
        className="photoWrapper"
      >
        <img
          className="photo"
        />
      </div>
      <div
        className="inform"
      >
        <p
          className="state alive"
        >
          Alive
        </p>
        <p
          className="name"
        />
        <p>
          Дата рождения:
           
          <span
            className="birthday"
          >
             
          </span>
        </p>
        <p>
          Ник-нейм:
           
          <span
            className="nickname"
          >
             
          </span>
        </p>
        <p
          className="quoteMargin"
        >
          Цитата: 
          <span
            className="quote"
          >
             
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
